### PR TITLE
[Slate, Slate-React] Add commands to Slate React 

### PR DESCRIPTION
**Pull request by hanna-greaves**
Commands that were added to Slate Editor should also be available from Slate-React editor. This PR adds an interface for both describing the available commands ands implements the interface for each class.

This change was created based on [feedback from a previous PR](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/30163#issuecomment-437170271).

Please fill in this template.

----

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: 
[example demo code using the react component like this](https://github.com/ianstormtaylor/slate/blob/0c94ccebce18bb05eafb4466737adc6a576d1db9/examples/rich-text/index.js#L260)

### CODE REVIEW

**Comment by sgreav** [comment]
Please wait before merging, I would like to see what other maintainers think of this change

File: types/slate/index.d.ts
Comment (kalley) [inline types/slate/index.d.ts]: anything against this being `Controller` instead? Since it also holds queries (or should: I haven't looked at it yet). Really what I think this should be is a representations of the core plugins

Comment (sgreav) [reply types/slate/index.d.ts]: Ah yeah I think `Controller` is probably a better shout, I also need to move over the additional query commands to that interface, will do that

Comment (sgreav) [reply types/slate/index.d.ts]: I've push some changes to this effect

File: types/slate-react/index.d.ts
Comment (elliottsj) [inline types/slate-react/index.d.ts]: Thoughts on declaring all of these methods using the existing types on `EditorController`? e.g.

```ts
addMark: Controller['addMark'];
delete: Controller['delete'];
// etc
```

I think this will help to keep the two classes in sync as command signatures change over time, with the `EditorController` as the source of truth.

Comment (sgreav) [reply types/slate-react/index.d.ts]: I quite like this idea, I'll give it a shot

Comment (sgreav) [reply types/slate-react/index.d.ts]: I've updated the PR to use this syntax, thanks for the suggestion!

**Comment by sgreav** [comment]
I think this is ready for merge now, any maintainer please go ahead and merge this in when you get a chance

**Comment by sgreav** [comment]
@kalley @elliottsj, if you guys are okay with this change a sign off would be great :)

**Review by elliottsj — APPROVED** [review]
lgtm

**Pull request merged**
